src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -246,6 +257,37 @@ public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, in
             }
         }
 
+        private bool TryNavigateToMappedFile(Workspace workspace, ISpanMappingService spanMappingService, Document generatedDocument, TextSpan textSpan)
+        {
+            var vsWorkspace = workspace as VisualStudioWorkspaceImpl;
+            Contract.ThrowIfNull(vsWorkspace);
+            var results = ThreadingContext.JoinableTaskFactory.Run(async () =>
+            {
+                return await spanMappingService.MapSpansAsync(generatedDocument, SpecializedCollections.SingletonEnumerable(textSpan), CancellationToken.None).ConfigureAwait(true);
```

Comment:
I think we were requiring these methods to be free-threaded so the JTF isn't necessary. It won't harm anything other than somebody down the road trying to figure out what the contract is.